src/pubsub/mod_pubsub.erl

```diff
@@ -1396,23 +1392,48 @@ iq_action_to_metric_name(<<"purge">>) -> purge;
 iq_action_to_metric_name(<<"subscriptions">>) -> subscriptions;
 iq_action_to_metric_name(<<"affiliations">>) -> affiliations.
 
-
-metric_name(IQType, Name, MetricSuffix) when is_binary(Name) ->
+event_name(IQType, Name) when is_binary(Name) ->
     NameAtom = iq_action_to_metric_name(Name),
-    metric_name(IQType, NameAtom, MetricSuffix);
-metric_name(IQType, Name, MetricSuffix) when is_atom(Name) ->
-    [pubsub, IQType, Name, MetricSuffix].
-
-report_iq_action_metrics_before_result(Host, IQType, Name) ->
-    mongoose_metrics:update(Host, metric_name(IQType, Name, count), 1).
-
-report_iq_action_metrics_after_return(Host, Result, Time, IQType, Name) ->
-    case Result of
-        {error, _} ->
-            mongoose_metrics:update(Host, metric_name(IQType, Name, erros), 1);
-        _ ->
-            mongoose_metrics:update(Host, metric_name(IQType, Name, time), Time)
-    end.
+    event_name(IQType, NameAtom);
+event_name(set, create) ->
+    mod_pubsub_set_create;
+event_name(set, publish) ->
+    mod_pubsub_set_publish;
+event_name(set, retract) ->
+    mod_pubsub_set_retract;
+event_name(set, subscribe) ->
+    mod_pubsub_set_subscribe;
+event_name(set, unsubscribe) ->
+    mod_pubsub_set_unsubscribe;
+event_name(get, items) ->
+    mod_pubsub_get_items;
+event_name(get, options) ->
+    mod_pubsub_get_options;
+event_name(set, options) ->
+    mod_pubsub_set_options;
+event_name(get, configure) ->
+    mod_pubsub_get_configure;
+event_name(set, configure) ->
+    mod_pubsub_set_configure;
+event_name(get, default) ->
+    mod_pubsub_get_default;
+event_name(set, delete) ->
+    mod_pubsub_set_delete;
+event_name(set, purge) ->
+    mod_pubsub_set_purge;
+event_name(get, subscriptions) ->
+    mod_pubsub_get_subscriptions;
+event_name(set, subscriptions) ->
+    mod_pubsub_set_subscriptions;
+event_name(get, affiliations) ->
+    mod_pubsub_get_affiliations;
+event_name(set, affiliations) ->
+    mod_pubsub_set_affiliations.
+
+ip_action_result_to_measurements(_Time, {error, _}, From) ->
```

Comment:
Shouldn't it be `iq_action_...`?

Reply:
Of course... I don't know how it happened, and then it slipped me. Thanks for noticing, I'll fix it right away 👍